tools/validate/validate.cpp

```diff
@@ -191,7 +193,19 @@ int main() {
             continue;
         }
 
-        const string extension = rdi->path().extension().string();
+        const wstring& relative_path = filepath.native();
+
+        constexpr size_t maximum_relative_path_length = 120;
+        if (relative_path.size() > maximum_relative_path_length) {
+            fwprintf(stderr, L"Validation failed: the path \"%s\" is too long (%zu characters; the limit is %zu).\n",
+                filepath.c_str(), relative_path.size(), maximum_relative_path_length);
+        }
+
+        if (relative_path.find(L' ') != wstring::npos) {
+            fwprintf(stderr, L"Validation failed: the path \"%s\" contains spaces.\n", filepath.c_str());
+        }
+
+        const wstring extension = filepath.extension().wstring();

```

Comment:
`native()` ?

Reply:
I consider that to be dangerous. The issue is that `.extension()` returns a prvalue `path`. The signature of `native()` is `const string_type& native() const noexcept`, i.e. it can be called on a prvalue and return an lvalue reference to the guts of a temporary. Storing that in `const wstring` would be fine, but storing it in `const wstring&` would result in a dangling reference without lifetime extension.

For this reason, I use `wstring()` everywhere for prvalue `path`. The only place I use `native()`, stored in `const wstring&`, is when we have a long-lived `path` (specifically the one within the `recursive_directory_iterator`).

Reply:
Does it make any difference other than whether the caller or the callee performs the copy? We can't store a reference here since `extension()` is a prvalue.

Reply:
I think Casey is agreeing with me, but just to clarify for all:
`const wstring& extension = filepath.extension().native();` would be a dangling reference.
`const wstring extension = filepath.extension().native();` is correct but I regard it as one millimeter away from doom. (Copy performed in caller.)
`const wstring extension = filepath.extension().wstring();` is equally efficient, but doesn't risk doom. (Copy performed in callee.)